### DISCUSSION TITLE
🙋‍♀️🙋‍♂️ Questsions and answers from the Q&A sessions on April 16th

### DISCUSSION THREAD

**Discussion by Victorlouisdg** [Victorlouisdg/cloth-competition#19] [General]
# Cloth Competition Q&A on April 16th
## Question 1
What happens if the cloth gets tangled? 

**Answer:** The competition judges can allow a retry if the failure was not the participants' fault.
## Question 2
Will the data be collected at ICRA?

**Answer:** Yes, the same data as in the training dataset: the observation_start, observation_result, and grasp folders, will also be collected during the competition. This data will also be made publicly available.
## Question 3
Will we receive a performance metric?

**Answer:** Yes, we are working on that right now. You can already view the pull request on the Github repository. The metric will be coverage as explained in the rules of the competition. We are working on a user interface where we can validate masks extracted by segment anything to calculate the coverage.
## Question 4
What will the range of sizes of the towels and shirts be?

**Answer:** We will not deviate much from the garment sizes in the training dataset. For reference, the largest towels are approximately 70 x 50 cm. 
## Question 5
How will we get the data at ICRA?

**Answer:** We are working on an HTTP-like server where users will be able to GET the latest observations and POST their desired grasp. More details about this will follow soon.
## Question 6
How will the 30-second time for grasp prediction be measured?

**Answer:** We will start measuring the time when we receive the GET request from the team for the start observation. In the subsequent 30 seconds, teams can send their grasp.
## Question 7
What happens if a team cannot complete all trials within its time slot? Will you average the result of the trial that did succeed within the time slot?

**Answer:** We will ensure that all teams execute the same number of trials. All teams will have to compete with the same set of items to ensure fairness, as some items can be more difficult than others. We will schedule buffer time in case teams cannot complete all trials within their time slot.
## Question 8
How will you handle the cloth slipping from the gripper when stretching? Will you consider the surface just before slipping or will the result after slipping count?

**Answer:** We will use the image after slipping, participants have the responsibility to provide secure grasps. Note that we have taken additional measures to make slipping rare.

One of the measures we have taken is reducing the tension threshold at which stretching stops to 2 Newton (instead of 4). Additionally, there was only 1 very low-friction towel in the training set where slipping was a significant problem. We will not include this towel for the evaluation at ICRA. For the t-shirts, slipping was rarely a problem due to the high-friction fabric they are made of. 

In the rare event that slipping would still occur during the evaluation, we will also consider which gripper the cloth slipped from. If the cloth slips from the left gripper (the one the participants did not control), the participant will be allowed to retry. If the cloth slips from the right gripper (the one that executed the participant’s grasp), then the participant cannot retry. 
## Question 9
Will you livestream the competition?

**Answer:** We have not planned a live stream yet, but are considering the possibility.

## Question 10
We noticed that some grasp poses are not reachable. How will it be handled during the competition when a participant sends a non-reachable pose?

**Answer:** When a team sends a grasp pose, we try running our motion planning toward that pose (this takes a few seconds). If a solution is found, the grasp is immediately executed. If no solution is found, we will inform the team and they can send additional poses within the 30-second time window.

## Question 11
Will the receiving/sending data to the competition server be the same during the remote dry run as at ICRA 2024?

**Answer:** Yes, the server will have the same API, only the IP address should be different.   

## Question 12 
Can we use hardcoded pixel coordinates for bounding boxes e.g. to segment cloth?

**Answer:** Yes, but it is safer to use 3D coordinates. For example, you can use the 3D TCP poses of the robots included in the observations. I will try to mimic the camera position from the lab as closely as possible at ICRA, but it will not be exactly the same.
